starlite/middleware/logging.py

```diff
@@ -0,0 +1,324 @@
+            if not hasattr(self, "logger"):
+                self.logger = scope["app"].get_logger(self.config.logger_name)
+                self.is_struct_logger = structlog_installed and isinstance(self.logger, BindableLogger)
+            if self.config.request_log_fields:
+                await self.log_request(scope=scope)
+            if self.config.response_log_fields:
+                send = self.create_send_wrapper(scope=scope, send=send)
+        await self.app(scope, receive, send)
+
+    async def log_request(self, scope: "Scope") -> None:
+        """
+        Handles extracting the request data and logging the message.
+        Args:
+            scope: The ASGI connection scope.
+
+        Returns:
+            None
+
+        """
+        extracted_data = await self.extract_request_data(request=Request[Any, Any](scope))
+        self.log_message(values=extracted_data)
+
+    def log_response(self, scope: "Scope") -> None:
+        """Handles extracting the response data and logging the message.
+
+        Args:
+            scope: The ASGI connection scope.
+
+        Returns:
+            None
+        """
+        extracted_data = self.extract_response_data(scope=scope)
+        self.log_message(values=extracted_data)
+
+    def log_message(self, values: OrderedDict[str, Any]) -> None:
+        """
+
+        Args:
+            values: Extract values to log.
+
+        Returns:
+            None
+
+        """
+        message = values.pop("message")
+        if self.is_struct_logger:
+
+            self.logger.info(message, **values)
+        else:
+            self.logger.info(f"{message}: " + ", ".join([f"{key}={value}" for key, value in values.items()]))
+
+    async def extract_request_data(self, request: "Request") -> OrderedDict[str, Any]:
+        """Creates a dictionary of values for the message.
+
+        Args:
+            request: A [Request][starlite.connection.Request] instance.
+        Returns:
+            An OrderedDict.
+        """
+
+        data: OrderedDict[str, Any] = OrderedDict([("message", self.config.request_log_message)])
+        extracted_data = self.request_extractor(connection=request)
+        serializer = get_serializer_from_scope(request.scope) or default_serializer
+        for key in self.config.request_log_fields:
+            if key in extracted_data:
+                value = extracted_data[key]  # pyright: ignore
+                if request.method == HttpMethod.GET and key == "body":
+                    continue
+                if isawaitable(value):
+                    value = await value
+                if not self.is_struct_logger and isinstance(value, (dict, list, tuple, set)):
+                    value = dumps(value, default=serializer, option=OPT_SERIALIZE_NUMPY | OPT_OMIT_MICROSECONDS)
+                if isinstance(value, bytes):
+                    value = value.decode("utf-8")
+                data[key] = value
```

Comment:
```suggestion
        for key in self.config.request_log_fields:
            if key not in extracted_data:
                continue
            value = extracted_data[key]  # pyright: ignore
            if request.method == HttpMethod.GET and key == "body":
                continue
            if isawaitable(value):
                value = await value
            if not self.is_struct_logger and isinstance(value, (dict, list, tuple, set)):
                value = dumps(value, default=serializer, option=OPT_SERIALIZE_NUMPY | OPT_OMIT_MICROSECONDS)
            if isinstance(value, bytes):
                value = value.decode("utf-8")
            data[key] = value
```

Reply:
I see. 

Tbh it should be actually assert, because it's impossible for the value to be missing.

Reply:
Assertions in production code... 🙃 

Maybe the check should just be removed?

Reply:
done